### PR TITLE
added elections in foundation + other stages where needed

### PR DESCRIPTION
**Pull request by lucianecchi**
Fixed bugs: 
- elections weren't ran in foundation stage 
- gamestate wasn't updated before elections in other phases (such as after kicking out the leader)

### CODE REVIEW

File: internal/server/RoundLoop.go [hunk 1]
```diff
@@ -45,7 +46,24 @@ func (s *Server) RunRoundLoop() {
 	s.punishBikelessAgents()
 
 	// Check if agents died
+	// Check Audi collision
+	s.AudiCollisionCheck()
 	s.unaliveAgents()
+	s.UpdateGameStates()
+
+	// if the leader dies hold new elections
```
Comment (SOMAS2023) [inline internal/server/RoundLoop.go:54]: Well remembered!

File: internal/server/RoundLoop.go [hunk 2]
```diff
@@ -77,48 +95,56 @@ func (s *Server) RunBikeSwitch(gameState GameStateDump) {
...
-			dictator := s.GetAgentMap()[bike.GetRuler()]
-			agentsVotes = dictator.DecideKickOut()
-		}
+			case utils.Leadership:
+				// get the map of weights from the leader
+				ruler := bike.GetRuler()
+				fmt.Printf("the ruler is %s \n", ruler)
+				leader := s.GetAgentMap()[bike.GetRuler()]
+				weights := leader.DecideWeights(utils.Kickout)
+				// get which agents are getting kicked out
```
Comment (SOMAS2023) [inline internal/server/RoundLoop.go:122]: I suppose leader could be malevolent and never get kicked out if he sets all other agent weights to zero

**Review by SOMAS2023 — APPROVED** [review]
Looks great @lucianecchi! Thanks for doing all this, you're a hero

File: internal/server/RoundLoop.go (hunk 2 above)
Comment (lucianecchi) [reply internal/server/RoundLoop.go:122]: mhh yeah I think that coudl be an interesting behaviour

**Pull request merged**